File: Chess.Game/Pieces/IPiece.cs
using System.Collections.Generic;

namespace Chess.Game.Pieces
{
    public interface IPiece
    {
        int Moves { get; set; }
        Color Color { get; set; }

        IList<Location> GetLegalMoves(Location currentLocation);
        bool IsMoveLegal(Location current, Location newLocation);
    }
}

File: Chess.Web/ClientApp/pieces/IPiece.ts
import { Location } from '../store/Game';
import Color from './Color';

interface IPiece {
    moved: boolean;
    color: Color;
    // file: number;
    // rank: number;
    isMoveLegal(current: Location, newLocation: Location, board: Location[][]): boolean;
    getImageName(): string;
}

export default IPiece;